nautobot/ipam/models.py

```diff
@@ -425,18 +470,141 @@ def family(self):
             return self.prefix.version
         return None
 
-    def get_duplicates(self):
-        return Prefix.objects.net_equals(self.prefix).filter(vrf=self.vrf).exclude(pk=self.pk)
+    def reparent_subnets(self):
+        """Determine the list of child nodes and set the parent to self."""
+        return self.subnets(direct=True, for_update=True).update(parent=self)
+
+    def supernets(self, direct=False, include_self=False, for_update=False):
+        """
+        Return supernets of this Prefix.
+
+        Args:
+            direct (bool): Whether to only return the direct ancestor.
+            include_self (bool): Whether to include this Prefix in the list of supernets.
+            for_update (bool): Lock rows until the end of any subsequent transactions.
+
+        Returns:
+            QuerySet
+        """
+        query = Prefix.objects.all()
+
+        if for_update:
+            query = query.select_for_update()
+
+        if direct:
+            return query.filter(id=self.parent_id)
+
+        if not include_self:
+            query = query.exclude(id=self.id)
+
+        return query.filter(
+            ip_version=self.ip_version,
+            prefix_length__lte=self.prefix_length,
+            network__lte=self.network,
+            broadcast__gte=self.broadcast,
+            # namespace=self.namespace,
+        )
```

Comment:
It's implied by meta ordering, so that there is always natural sorting by `network, `broadcast`, `prefix_length`. Once we add `namespace` that will be the first one in the list.

I will document. Thanks.